core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -357,7 +358,7 @@ public ByteBuffer keyMetadata() {
 
   @Override
   public List<Long> splitOffsets() {
-    return splitOffsets;
+    return splitOffsets != null ? Lists.newArrayList(splitOffsets) : null;
```

Comment:
Should this return an `ImmutableList` so that it cannot be modified? That's what was happening before, although I think it matter less if this is creating a new list each time it is returned.

Reply:
I thought that we use unmodified collection before because we don't wanna to change the contents inside splitOffsets.  Now we've accomplished the same purpose.  Returning it as modifiable or unmodifiable collection, both of them sounds good to me.